src/utils/utils_posix_common.c

```diff
@@ -266,3 +266,27 @@ int utils_file_open_or_create(const char *path) {
 
     return fd;
 }
+
+// Expected input:
+// char *str_threshold = utils_env_var_get_str("UMF_PROXY", "size.threshold=");
+size_t utils_get_size_threshold(char *str_threshold) {
+    if (!str_threshold) {
+        return 0;
+    }
+
+    // move to the beginning of the number
+    str_threshold += strlen("size.threshold=");
+
+    long long_threshold = atol(str_threshold);
+    if (long_threshold < 0) {
+        LOG_ERR("size threshold cannot be negative! = (char *) %s, (int) %li",
+                str_threshold, long_threshold);
+        return 0;
```

Comment:
jus confirming: do we want to return 0, like nothing happened? if we accept 0 as a proper result of this function then we should just set `long_threshold=0` and let it print in the debug msg below, perhaps...?

Reply:
Changed to `-1` - error.
Done.